Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "BenchmarkGymGameModeBase.h"
+
+#include "Engine/World.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GDKTestGymsGameInstance.h"
+#include "Interop/SpatialWorkerFlags.h"
+#include "Misc/CommandLine.h"
+#include "Utils/SpatialMetrics.h"
+
+DEFINE_LOG_CATEGORY(LogBenchmarkGymGameModeBase);
+
+const FString ABenchmarkGymGameModeBase::TotalPlayerWorkerFlag = TEXT("total_players");
+const FString ABenchmarkGymGameModeBase::TotalNPCsWorkerFlag = TEXT("total_npcs");
+const FString ABenchmarkGymGameModeBase::TotalPlayerCommandLineKey = TEXT("TotalPlayers");
+const FString ABenchmarkGymGameModeBase::TotalNPCsCommandLineKey = TEXT("TotalNPCs");
+
+// Metrics
+namespace
+{
+	const FString AverageClientRTTMetricName = TEXT("UnrealAverageClientRTT");
+	const FString AverageClientViewLatenessMetricName = TEXT("UnrealAverageClientViewLateness");
+	const FString PlayersSpawnedMetricName = TEXT("UnrealActivePlayers");
+
+	const FString MaxRoundTripWorkerFlag = TEXT("max_round_trip");
+	const FString MaxLatenessWorkerFlag = TEXT("max_lateness");
+	const FString MaxRoundTripCommandLineKey = TEXT("MaxRoundTrip");
+	const FString MaxLatenessCommandLineKey = TEXT("MaxLateness");
+
+} // anonymous namespace
+
+ABenchmarkGymGameModeBase::ABenchmarkGymGameModeBase()
+	: ExpectedPlayers(1)
+	, PrintUXMetric(10.0f)
+	, MaxClientRoundTripSeconds(150)
+	, MaxClientViewLatenessSeconds(150)
+	, bPlayersHaveJoined(false)
+	, bHasUxFailed(false)
+	, bHasFpsFailed(false)
+	// These values need to match the GDK scenario validation equivalents
+	, MinAcceptableFPS(20.0f) // Same for both client and server currently
+	, MinDelayFPS(120.0f)
+	, ActivePlayers(0)
+{
+	PrimaryActorTick.bCanEverTick = true;
+	SecondsTillPlayerCheck = 15.0f * 60.0f;
+
+	// Seamless Travel is not currently supported in SpatialOS [UNR-897]
+	bUseSeamlessTravel = false;
```

Comment:
Do we use this?

Reply:
Doesn't seem so, I just kept it from what was there.

Reply:
Ah might as well remove, it was probably some legacy code that has gone now!